components/style/restyle_hints.rs

```diff
@@ -573,6 +574,23 @@ pub struct DependencySet {
     dependencies: SelectorMap<Dependency>,
 }
 
+/// The data that we need to compute a given restyle hint.
+pub enum HintComputationContext<'a, E: 'a>
+    where E: TElement,
+{
+    /// The data we need to compute a restyle hint for the root element.
```

Comment:
This comment needs to change before landing, saying something like "for the root of the traversal".